openmeter/productcatalog/plan/validator.go

```diff
@@ -0,0 +1,17 @@
+package plan
+
+type Validator interface {
+	// Validate returns an error if the instance of the Validator is invalid.
+	Validate() error
+}
+
+type Equaler[T any] interface {
+	// StrictEqual must return true in case all attributes of T are strictly equal.
+	StrictEqual(T) bool
+
+	// Equal is a relaxed version of StrictEqual where it is allowed to return true in case a subset of the attributes are equal.
+	// Typically, you want to use Equal to compare two instances of T where one or both missing some metadata which are
+	// not required for comparison. Like comparing two instances of T before and after stored in data layer which assigns
+	// additional metadata like timestamps, unique identifiers, etc.
+	Equal(T) bool
```

Comment:
just an opinion, but I'm not sure I'm sold on this... if this behaves differently case-by-case then IMO it should be a different thing altogether

Reply:
I'll remove it for now and we can discuss it separately. This is not required for this PR.